TypeCobol/Compiler/Scanner/Scanner.cs

```diff
@@ -1445,9 +1446,53 @@ currentIndex < (lastIndex+1) && (line[currentIndex + 2] == '"' || line[currentIn
                     int saveCurrentIndex = currentIndex;
                     Token numericLiteralToken = ScanNumericLiteral(startIndex);
 
+                    //TODO It's a guess, it need to be tested
+                    if (numericLiteralToken.TokenType == TokenType.FloatingPointLiteral)
+                    {
+                        //a FloatingPointLiteral contains a character `.` which is invalid in a UserDefinedWord or a keyword, so we can return this token directly.
+                        return numericLiteralToken;
+                    }
+
+                    //TODO it's a basic code to handle #2167, but it seems easy to handle all UserDefinedWord with the form 123-456
+                    //Paragraph and section with name like "123" could also be handled this way
                     // 2. Then check to see if the next char would be valid inside a CobolWord
+                    bool nextCharIsDashFollowedByCobolChar = currentIndex < lastIndex && line[currentIndex] == '-' && CobolChar.IsCobolWordChar(line[currentIndex + 1]);
                     bool nextCharIsACobolWordChar = (currentIndex <= lastIndex) && CobolChar.IsCobolWordChar(line[currentIndex]);
-                    if(nextCharIsACobolWordChar && line[currentIndex] == '-')
+                    if (currentState.LastSignificantToken != null && (currentState.LastSignificantToken.TokenType == TokenType.PERFORM || currentState.LastSignificantToken.TokenType == TokenType.THRU || currentState.LastSignificantToken.TokenType == TokenType.THROUGH || currentState.LastSignificantToken.TokenType == TokenType.OF || currentState.LastSignificantToken.TokenType == TokenType.IN)
```

Comment:
So this is for COBOL only not TypeCobol, otherwise TokenType.QualifiedNameSeparator should be possible here.

Reply:
Yes I forgot TypeCobol.
But as discussed before, we'll now handle #1982, so I'll remove this test.

Reply:
This code has been removed and is now generic for Cobol and TypeCobol.